src/main/java/duke/gui/DialogBox.java

```diff
@@ -0,0 +1,115 @@
+package duke.gui;
+
+import java.io.IOException;
+import java.util.Collections;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.geometry.Pos;
+import javafx.scene.Node;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.HBox;
+import javafx.scene.paint.Color;
+import javafx.scene.shape.Rectangle;
+import javafx.scene.text.Text;
+
+/**
+ * A custom control using FXML that handles the Dialogue Box in the GUI.
+ * It represents a Dialogue Box consisting of an ImageView to represent the speaker's face
+ * and a label containing text from the speaker.
+ * Reused from https://se-education.org/guides/tutorials/javaFx.html
+*/
+public class DialogBox extends HBox {
+
+    private static final float WRAPPING_WIDTH = 250;
+    private static final float IMAGE_LENGTH = 70;
+    private static final float ROUNDED_CORNERS_LENGTH = 25;    
+
+    private static final Color DUKE_COLOR = Color.CORNFLOWERBLUE;
+    private static final Color USER_COLOR = Color.CRIMSON;
```

Comment:
Perhaps associated constants should have a constant prefix?

Reply:
Thanks for pointing this out! I didn't notice this before now.